### PR TITLE
Update Contentful documentation around the developer experience within Bedrock

### LINKED ISSUE #11389: Document Contentful dev workflow
## Description

We have [Contentful documentation](https://bedrock.readthedocs.io/en/latest/contentful.html) that describes the system and components, but that doesn't yet include any real information on the developer workflow and setup that's required to add or make new changes to both bedrock and contentful.

Things that would be useful to explain:

- How to get bedrock set up locally to work with Contentful (required env variables).
- How to preview your changes on `localhost` when working on a feature branch.
- How to make changes to the `sandbox` environment to make sure your code works as expected.
- Anything else? (maybe backend specific?)

---

#### Success Criteria
- [ ] Contentful developer workflow is documented.

### PR DESCRIPTION
**Pull request by stevejalim**
Resolves #11389

### CODE REVIEW

File: docs/contentful.rst [hunk 1]
```diff
@@ -379,6 +385,71 @@ It's not problematic to have them created by non developers, it's just that they
...
+``CONTENTFUL_NOTIFICATION_QUEUE_URL``
+``CONTENTFUL_NOTIFICATION_QUEUE_REGION``
+``CONTENTFUL_NOTIFICATION_QUEUE_ACCESS_KEY_ID``
+``CONTENTFUL_NOTIFICATION_QUEUE_SECRET_ACCESS_KEY``
+
+
+How to preview your changes on localhost
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When viewing a page in Contentful, it's possible to trigger a preview of the draft page. This is typically rendered on www-dev.allizom.org. However, that's only useful for code that's already in ``main``.
+If you want to preview Contentful content on your local machine - eg you're working on a feature branch that isn't ready for merging - do the following:
```
Comment (robhudson) [inline docs/contentful.rst:416]: I would expand "eg" to it's full form, "e.g."

File: docs/contentful.rst [hunk 2]
```diff
@@ -379,6 +385,71 @@ It's not problematic to have them created by non developers, it's just that they
...
+
+Note that previewing a page will require it to be pulled from Contentful's API, so you will need ``CONTENTFUL_SPACE_ID`` and ``CONTENTFUL_SPACE_KEY`` set in your ``.env``. It may take a few seconds to get the data.
+
+Also note that when you select ``Localhost preview``, the choice sticks, so you should set it back to ``Preview on web`` when you're done.
+
+
+How to update/refresh the sandbox environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It helps to think of Contentful 'environments' as simply branches of a git-like repo full of content. You can take a particular environment and branch off it to make a new environment for WIP or experimental content, using the original one as your starting point.
```
Comment (robhudson) [inline docs/contentful.rst:431]: Can we add a `:abbr:` for "WIP"?

File: docs/contentful.rst [hunk 3]
```diff
@@ -379,6 +385,71 @@ It's not problematic to have them created by non developers, it's just that they
...
+On top of this, Contentful has the concept of aliases for environments and we use two aliases in our setup:
+
+* ``master`` is used for production and is an alias currently pointing to the `V1` environment. It is pretty stable and access to it is limited.
+* ``sandbox`` is used for development and more team members have access to edit content. Again, it's an alias and is pointed at an environment (think, branch) with a name in the format ``sandbox-YYYY-MM-DD``.
+
+
+While updating ``master`` is something that we generally don't do (at the moment only Dan B would do this), updating the sandbox happens more often, typically to populate it with data more recently added to master.
```
Comment (robhudson) [inline docs/contentful.rst:438]: Instead of a person's name can we say something like "Contentful admin" or similar?

File: docs/contentful.rst [hunk 4]
```diff
@@ -379,6 +385,71 @@ It's not problematic to have them created by non developers, it's just that they
...
+* Ensure we have at least one spare environment slot. If we don't delete the oldest ``sandbox-XXXX-XX-XX`` environment.
+* Click the blue Add Environment button, to the right. Name it using the ``sandbox-YYYY-MM-DD`` pattern and base it on whatever environment is aliased to ``master`` - this will basically create a new 'branch' with the content currently in master.
+* In the Environment Aliases section of the main page, find `sandbox` and click Change alias target, then select the ``sandbox-XXXX-XX-XX`` environment you just made.
+
+Which environment is connected to where?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``master`` is the environment used in Bedrock production, stage, dev and test
+``sandbox`` may, in the future, be made the default environment for dev. It's also the one we should use for local development.
+
+If you develop a new feature that adds to Contentful (eg page or component) and you author it in the sandbox, you will need to re-create it in master before the corresponding bedrock changes hit production.
```
Comment (robhudson) [inline docs/contentful.rst:450]: `s/eg/e.g./` ?

**Review by robhudson — APPROVED** [review]
r+wc

Good docs, it helped me understand how contentful is set up.

Aside: Is it possible to change the name of the "master" environment in Contentful to match our recent change of default branch in git?

**Comment by stevejalim** [comment]
> Aside: Is it possible to change the name of the "master" environment in Contentful to match our recent change of default branch in git?

Yes! Definitely. This is on my follow-up list and I'll sort it out now

**Pull request merged**